### PR TITLE
pythonPackages update: asn1ate and asn2quickder

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Upgrade to latest released version of these asn1 tools.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/asn1ate/default.nix [hunk 1]
```diff
@@ -3,13 +3,12 @@
 
 buildPythonPackage rec {
   pname = "asn1ate";
-  date = "20160810";
-  version = "unstable-${date}";
+  version= "0.6";
   name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/asn1ate/default.nix]: You can delete the `name`.

File: pkgs/development/tools/asn2quickder/default.nix [hunk 2]
```diff
@@ -1,30 +1,22 @@
-{ stdenv, fetchFromGitHub, python2Packages, makeWrapper }:
+{ stdenv, fetchFromGitHub, python2Packages, makeWrapper, cmake }:
 
-stdenv.mkDerivation rec {
+python2Packages.buildPythonPackage rec {
   pname = "asn2quickder";
   name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/development/tools/asn2quickder/default.nix]: Leave out name here as well.

File: pkgs/development/tools/asn2quickder/default.nix [hunk 3]
```diff
@@ -1,30 +1,22 @@
-{ stdenv, fetchFromGitHub, python2Packages, makeWrapper }:
+{ stdenv, fetchFromGitHub, python2Packages, makeWrapper, cmake }:
 
-stdenv.mkDerivation rec {
+python2Packages.buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/development/tools/asn2quickder/default.nix]: Better use `python2Packages.callPackage` and take `buildPythonPackage` as well as all dependencies as arguments.

File: pkgs/development/python-modules/asn1ate/default.nix (hunk 1 above)
Comment (leenaars) [reply pkgs/development/python-modules/asn1ate/default.nix]: Isn't that mandatory?

File: pkgs/development/tools/asn2quickder/default.nix (hunk 3 above)
Comment (leenaars) [reply pkgs/development/tools/asn2quickder/default.nix]: I don't understand what you mean by that? I thought his was a very common way of doing this.
Do you have a recommended example where this is done elsewhere in nixpkgs.
I grepped but could not find something that seems to fit here ...

File: pkgs/development/python-modules/asn1ate/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/asn1ate/default.nix]: Not for Python. `buildPythonPackage` will automatically derive ot from `pname` and `version`.

File: pkgs/development/tools/asn2quickder/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/tools/asn2quickder/default.nix]: In `all-packages.nix`, one can use `python2Packages.callPackage`. And in `default.nix` the first line should look something like
```
{ stdenv, fetchFromGitHub, python2Packages, makeWrapper, cmake, pytestrunner, pytest, six, pyparsing }:
```

Comment (dotlambda) [reply pkgs/development/tools/asn2quickder/default.nix]: Actually, this should be `buildPythonApplication`, since it's not a library.

**Comment by leenaars** [comment]
@dotlambda: I removed the 'name' but it turns out then you get a real ugly prefix, e.g. python2.7-asn1ate-0.6 and python2.7-asn2quickder-1.2-5 instead of asn1ate-0.6. So I reverted back to what I had.

**Comment by dotlambda** [comment]
Actually, I think this should be called `quick-der` now.
We can add `asn2quicker` as an alias.
But then, it doesn't make sense to use `buildPythonApplication`, since `quick-der` is much more than a python executable.
I don't have enough experience to know how to package C and Python libraries at the same time. Maybe @FRidh can help.

File: pkgs/development/tools/asn2quickder/default.nix (hunk 3 above)
Comment (leenaars) [reply pkgs/development/tools/asn2quickder/default.nix]: @dotlambda: When I do that, for some reason it tries to create a wheel and cannot find six. So it doesn't build anymore.

It builds perfectly fine as is. Would it not be okay to leave it as is? I only did a version bump...

**Comment by leenaars** [comment]
I don't know either, my solution was to keep them apart. That at least I understand and works... I have a separate quickder on my system which I will submit soon.

I don't think this should cost anyone a lot of time at this stage. When we would run into any issues later on, this could be fixed then. This is very useful but not mainstream technical tooling...

File: pkgs/development/tools/asn2quickder/default.nix (hunk 3 above)
Comment (leenaars) [reply pkgs/development/tools/asn2quickder/default.nix]: Thanks. Done.

File: pkgs/development/tools/asn2quickder/default.nix (hunk 2 above)
Comment (leenaars) [reply pkgs/development/tools/asn2quickder/default.nix]: Done.

File: pkgs/development/python-modules/asn1ate/default.nix (hunk 1 above)
Comment (leenaars) [reply pkgs/development/python-modules/asn1ate/default.nix]: Done. Thanks for helping out!

**Comment by leenaars** [comment]
All suggestions have been processed, should be ready to merge.

**Comment by Mic92** [comment]
@GrahamcOfBorg build quickder

**Comment by Mic92** [comment]
Was `quickder` broken before?

**Comment by FRidh** [comment]
`asn2quickder` builds on master but not with this commit. Some fixing is needed.

**Comment by leenaars** [comment]
I updated quickder to the latest release, should work now.

**Comment by FRidh** [comment]
@GrahamcOfBorg build quickder

**Comment by FRidh** [comment]
@leenaars quickder still seems to be failing.

**Comment by leenaars** [comment]
It builds on my NixOS and Debian systems. The failure is in both cases a script that is called which fails on:
/usr/bin/env: bad interpreter: No such file or directory

A little digging reveals that a chrooted system does not have /usr/bin/env by design.
(https://github.com/NixOS/nixpkgs/issues/6227)...

I used patchShebangs to hopefully fix the issue.

**Comment by leenaars** [comment]
@GrahamcOfBorg build quickder

**Comment by Mic92** [comment]
@GrahamcOfBorg build quickder

**Comment by Mic92** [comment]
@GrahamcOfBorg build quickder

**Pull request merged**